logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java

```diff
@@ -0,0 +1,72 @@
+package alluxio.logserver;
+
+import org.apache.log4j.Logger;
+import org.apache.log4j.spi.LoggerRepository;
+import org.apache.log4j.spi.LoggingEvent;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.io.ObjectInputStream;
+import java.net.Socket;
+
+public class AlluxioLog4jSocketNode implements Runnable {
+  private Socket mSocket;
+  private LoggerRepository mHierarchy;
+  private ObjectInputStream mObjectInputStream;
+
+  public AlluxioLog4jSocketNode(Socket socket, LoggerRepository hierarchy) {
+    mSocket = socket;
+    mHierarchy = hierarchy;
+    try {
+      mObjectInputStream = new ObjectInputStream(
+          new BufferedInputStream(mSocket.getInputStream())
+      );
+    } catch (InterruptedIOException e) {
+      Thread.currentThread().interrupt();
+    } catch (IOException e) {
+      e.printStackTrace();
+    } catch (RuntimeException e) {
+      e.printStackTrace();
+    }
+  }
+  @Override
+  public void run() {
+    LoggingEvent event;
+    Logger remoteLogger;
+    try {
+      if (mObjectInputStream != null) {
+        while (true) {
+          event = (LoggingEvent) mObjectInputStream.readObject();
+          remoteLogger = mHierarchy.getLogger(event.getLoggerName());
```

Comment:
is remoteLogger effective level static once `mHierarchy` is set?  can you move it out of the  loop?